### ISSUE TITLE
Full histogram for multi-pcap Pool not rendered when times are far apart

### ISSUE THREAD

**Issue by philrz** [brimdata/brim#1699]
Repro is with Brim commit a6d1e12, which uses Zed commit `72cdbed`.

Please bear with me as it takes a fair amount of test data to explain this one! Test pcaps can all be downloaded from:

* https://storage.googleapis.com/brimsec-public-test-data/wrccdc.pcap
* https://storage.googleapis.com/brimsec-public-test-data/ssl1.pcapng
* https://storage.googleapis.com/brimsec-public-test-data/ssl2.pcapng

One of our goals with the next Brim release will be to allow users to incrementally add data to the same Pools via `zapi` or `brimcap`, so that's what was being tested here when I found the issue.

Starting from a fresh Dev app install, I create new Pools with intent to store data for each pcap separately as well as a couple for combined pairings.

```
$ zdeps/zapi create -p wrccdc
pool created: wrccdc

$ zdeps/zapi create -p ssl1
pool created: ssl1

$ zdeps/zapi create -p ssl2
pool created: ssl2

$ zdeps/zapi create -p wrccdc-ssl1
pool created: wrccdc-ssl1

$ zdeps/zapi create -p ssl1-ssl2
pool created: ssl1-ssl2
```

Now I do the Brimcap loads to the respective pools.

```
$ zdeps/brimcap load -p wrccdc -root "$(pwd)/run/data/brimcap-root" ~/pcap/wrccdc.pcap
100.0% 476MiB858KiB604B/476MiB858KiB604B records=670823 

$ zdeps/brimcap load -p ssl1 -root "$(pwd)/run/data/brimcap-root" ~/pcap/ssl1.pcapng 
100.0% 2MiB667KiB764B/2MiB667KiB764B records=268 

$ zdeps/brimcap load -p ssl2 -root "$(pwd)/run/data/brimcap-root" ~/pcap/ssl2.pcapng 
100.0% 1MiB850KiB224B/1MiB850KiB224B records=347 

$ zdeps/brimcap load -p wrccdc-ssl1 -root "$(pwd)/run/data/brimcap-root" ~/pcap/wrccdc.pcap 
100.0% 476MiB858KiB604B/476MiB858KiB604B records=450474 

$ zdeps/brimcap load -p wrccdc-ssl1 -root "$(pwd)/run/data/brimcap-root" ~/pcap/ssl1.pcapng 
100.0% 2MiB667KiB764B/2MiB667KiB764B records=268 

$ zdeps/brimcap load -p ssl1-ssl2 -root "$(pwd)/run/data/brimcap-root" ~/pcap/ssl1.pcapng 
100.0% 2MiB667KiB764B/2MiB667KiB764B records=268 

$ zdeps/brimcap load -p ssl1-ssl2 -root "$(pwd)/run/data/brimcap-root" ~/pcap/ssl2.pcapng 
100.0% 1MiB850KiB224B/1MiB850KiB224B records=347 
```

After all this, I do a **View > Reload** inside Brim to make sure all these Pool updates are visible in the app.

Let's start with the example that works correctly, then I'll show the problem. The pcaps for "ssl1" and "ssl2" were taken close together in time. Seen separately, they're each small "lumps" of logs.

![image](https://user-images.githubusercontent.com/5934157/120870738-86864700-c54e-11eb-91f4-fdab8347af78.png)

![image](https://user-images.githubusercontent.com/5934157/120870746-8d14be80-c54e-11eb-8a62-cb4073632350.png)

Then combined we see both "lumps" in the histogram, so that seems healthy.

![image](https://user-images.githubusercontent.com/5934157/120870764-9bfb7100-c54e-11eb-916f-8883c3e854ef.png)

However, the problem arises with the other pairing. The "wrccdc" pcap happened way back in 2018. Seen in its own Pool, it's a big lump.

![image](https://user-images.githubusercontent.com/5934157/120870782-ad447d80-c54e-11eb-9efc-b65d0cd8726c.png)

Now here's the problem: For the Pool where both the "wrccdc" pcap and "ssl1" were imported, we can see the "lump" for the ssl1 one far off to the right (a narrow sliver, which is unsurprising because the wide time span that starts back in 2018 due to the wrccdc data is going to make for narrow bars), but there's no corresponding "lump" on the left side where we'd expect to see the wrccdc data.

![image](https://user-images.githubusercontent.com/5934157/120870799-c0574d80-c54e-11eb-8f90-c912b651dd94.png)

I sniffed the data coming from the backend in Wireshark and it looks like the data to populate the histogram _is_ coming back. At that point I eyeballed the chart and noticed the leftmost X-axis label shows "April", which is a little suspicious given that the left side of the time picker due to the start of the wrccdc pcap data is in March. Indeed, if I manually manipulate the time picker and move that start back a month, now a lump appears, though it also causes me to lose my right-side sliver for the ssl1 data.

![image](https://user-images.githubusercontent.com/5934157/120870946-39ef3b80-c54f-11eb-8603-821fad1a1a51.png)

In conclusion, I suspect there's some kind of boundary problem with how the histogram is being rendered.

**Comment by philrz** [comment]
Verified in Brim commit 55390e8.

After repeating the original repro steps to import all the test data, the attached video shows my successful interactions with the Pool that previously exhibited the problem. As we can see, the "lump" that represents the much larger wrccdc data set appears as a single tall bar on the left side of the histogram. The second Pool contains so few records that it's effectively dwarfed and no longer has a visible bar. However, if I hover my pointer over that area of the histogram, the counts do appear. @mason-fish pointed out that we've had UX discussions at times about possible visual cues we could add to draw a user's attention to ranges in the histogram that do/don't have data, but I'll pursue that as a separate issue.

https://user-images.githubusercontent.com/5934157/122818599-4503d480-d28e-11eb-90a6-04fc1115d882.mp4

Thanks @mason-fish!

**Comment by philrz** [comment]
I just noticed another example where this fix improved things, even with just a single pcap's worth of data. Importing the attached [ifconfig.pcapng.gz](https://github.com/brimdata/brim/files/6717721/ifconfig.pcapng.gz) (after uncompressing) into the last GA Brim release `v0.24.0` and on up through commit c9c2e50 that was right before the changes in the linked PR, no bar was rendered at all for this data, presumably because the time band was so narrow.

![image](https://user-images.githubusercontent.com/5934157/123460130-062a9300-d59c-11eb-8306-a5c1642eb499.png)

Now in the time since Brim commit 2c8556a brought the changes in the linked PR, we do see a bar.

![image](https://user-images.githubusercontent.com/5934157/123460210-235f6180-d59c-11eb-9387-40c96d751f74.png)

Thanks again @mason-fish!
